distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Replace_Helpers.enso

```diff
@@ -0,0 +1,44 @@
+from Standard.Base import all
+
+import Standard.Base.Errors.Empty_Error.Empty_Error
+import Standard.Base.Errors.Illegal_Argument.Illegal_Argument
+
+import Standard.Database.Data.Table.Table as Database_Table
+
+import project.Data.Table.Table
+import project.Data.Set_Mode.Set_Mode
+from project.Errors import Missing_Input_Columns, No_Such_Column, Non_Unique_Key, Unmatched_Rows_In_Lookup
+
+replace : Table -> Table | Map -> (Text | Integer) -> (Text | Integer | Nothing) -> (Text | Integer | Nothing) -> Boolean -> Problem_Behavior -> Table ! Missing_Input_Columns | Non_Unique_Key | Unmatched_Rows_In_Lookup
+replace base_table:(Table | Database_Table) lookup_table:(Table | Database_Table | Map) column:(Text | Integer) from_column:(Text | Integer | Nothing)=Nothing to_column:(Text | Integer | Nothing)=Nothing allow_unmatched_rows:Boolean=True on_problems:Problem_Behavior=Problem_Behavior.Report_Warning =
+    case lookup_table of
+        _ : Map ->
+            if from_column.is_nothing.not || to_column.is_nothing.not then Error.throw (Illegal_Argument.Error "If a Map is provided as the lookup_table, then from_column and to_column should not also be specified.") else 
+                # An empty map is a no-op, but we attach a warning.
+                if lookup_table.is_empty then Warning.attach (Empty_Error.Error lookup_table) base_table else
+                    base_table.replace (base_table.make_table_from_map lookup_table 'from' 'to') column 'from' 'to' allow_unmatched_rows=allow_unmatched_rows on_problems=on_problems
```

Comment:
This is not necessarily true.

If `allow_unmatched_rows=False`, and if the lookup table is _empty_ and **if** the base table is _non-empty_, then this is not a no-op, but would fail with `Unmatched_Rows_In_Lookup`.

I know it is a rare edge case, but we need to handle it to satisfy the specification we have laid out in the documentation. It also makes sense intuitively - if I call `Table.replace` with `allow_unmatched_rows=False`, I expect _all my row values will be replaced_. So if I get an empty lookup table, my values are not replaced - in this context I will expect an error.

Please add a test for this edge case so we can make sure it works as expected.

Reply:
Done.